### PR TITLE
feat(devops): Add scripts to set up Bitcoin local node

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

We want to be able to test that the development of the Bitcoin integration works as expected.

In order to do that, we need a local Bitcoin node.

This PR provides two main scripts to set up the local node and start dfx with Bitcoin. As well as some other utilities to be used while testing.

# Changes

* Add a couple of entries to `.gitignore`.
* Add a script to set up the local Bitcoin node.
* Add a script to start dfx with Bitcoin.
* Add a script to mine tokens.
* Add a section in Hacking.md to explain how to use the scripts.

# Tests

I and Denys tested locally that the scripts worked as expected.

### CODE REVIEW

**Comment by lmuntaner** [comment]
@DenysKarmazynDFINITY We will look into this tomorrow to help you set up your environment.

**Comment by lmuntaner** [comment]
This PR is based on #2158 by @AntonioVentilii-DFINITY

**Comment by lmuntaner** [comment]
@peterpeterparker @bitdivine @AntonioVentilii-DFINITY this PR is ready for review.

@DenysKarmazynDFINITY went through the setup without any problems.

File: scripts/setup.bitcoin-node.sh [hunk 1]
```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+# Initialize a flag for the --reset option
+reset_flag=false
+
+# Loop through the arguments
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+  --reset)
```
Comment (bitdivine) [inline scripts/setup.bitcoin-node.sh:9]: Support for a `--help` flag is always welcome.

**Review by bitdivine — APPROVED** [review]
LGTM, thanks!

File: scripts/setup.bitcoin-node.sh (hunk 1 above)
Comment (lmuntaner) [reply scripts/setup.bitcoin-node.sh:9]: Good idea, done!

**Pull request merged**